lib/service_manager.dart

```diff
@@ -400,24 +400,36 @@ class ServiceExtensionManager {
     switch (expectedValueType) {
       case bool:
         final bool enabled = response.json['enabled'] == 'true' ? true : false;
-        await setServiceExtensionState(name, enabled, enabled,
-            callExtension: false);
+        if (_extensionEnabledOnDevice(name, enabled)) {
+          await setServiceExtensionState(name, enabled, enabled,
+              callExtension: false);
+        }
         return;
       case String:
         final String value = response.json['value'];
-        await setServiceExtensionState(name, true, value, callExtension: false);
+        if (_extensionEnabledOnDevice(name, value)) {
```

Comment:
I think this would all be clearer as a single helper that does all of
        if (_extensionEnabledOnDevice(name, enabled)) {
            callExtension: false);	          await setServiceExtensionState(name, enabled, enabled,
              callExtension: false);
        }
and inlines the confusingly named _extensionEnabledOnDevice method.

Reply:
done. also changed name to _maybeRestoreExtension.